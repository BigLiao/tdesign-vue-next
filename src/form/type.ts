export interface FormData {
  [key: string]: any;
}

export interface ErrorItem {
  field: string;
  name: string;
  message: string;
}

export type ErrorList = Array<ErrorItem>;

export interface ErrorObject {
  [key: string]: ValidateParams;
}

export type ValidateResult = boolean | ErrorObject;

export interface FormProps {
  data: Array<FormData>;
  labelAlign: 'left' | 'right' | 'top';
  layout: 'vertical' | 'inline';
  size: 'medium' | 'large';
  colon: boolean;
  requiredMark: boolean;
  scrollToFirstError: boolean;
  showErrorMessage: boolean;
  onValidate: (validateResult: ValidateResult) => void;
  onReset: () => void;
  onSubmit: () => void;
}

export interface FormItemProps {
  name: string;
  tooltip: string;
  label: string;
  for: string;
}

export type InnerValidator = 'required' | 'email' | 'date' | 'boolean' | 'number' | 'max' | 'min' | 'length' | 'idcard' | 'telnumber' | 'url' | 'enum' | 'pattern' | 'validator';

export type CustomValidator = (val: ValueType) => boolean | Promise<boolean>

export type ValidateRule = {
  validator: CustomValidator;
  message: string;
  trigger: 'change' | 'blur';
  type: 'error' | 'warning';
} & {
  [key in InnerValidator]: string;
}

export interface ValidateRules {
  [key: string]: Array<ValidateRule>;
}

export type ValueType = any;

export interface ValidateParams {
  [key: string]: any;
}

export interface ValidateOptions {
  field: string;
  value: ValueType;
  rules: Array<ValidateRule>;
}

export type CustomValidate = (val: ValueType) => boolean | Promise<boolean>;
