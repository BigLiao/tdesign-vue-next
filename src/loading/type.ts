export interface LoadingProps {
  loading: boolean;
  indicator?: () => void;
  text?: string;
  size?: string;
  delay?: number;
  fullscreen?: boolean;
  className?: string;
  showOverlay?: boolean;
  preventScrollThrough?: boolean;
  isService?: boolean;
}
