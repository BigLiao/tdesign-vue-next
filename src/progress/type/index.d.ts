
import { VNode } from 'vue/types/umd';

export interface ProgressProps {
  theme?: string;
  percentage?: number;
  label?: boolean |  (() => VNode) | TNode;
  status?: string;
  color?: string | any;
  trackColor?: string;
  strokeWidth?: string | number;
  size?: string;
  className?: ClassName;
}

export interface DiameterType {
  getDiameter?: number;
}
