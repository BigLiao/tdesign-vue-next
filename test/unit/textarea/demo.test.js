import { mount } from '@vue/test-utils';
import base from '@/examples/textarea/demos/base.vue';
import type from '@/examples/textarea/demos/type.vue';
import events from '@/examples/textarea/demos/events.vue';
import maxlength from '@/examples/textarea/demos/maxlength.vue';

describe('Textarea', () => {
  it('base demo works fine', () => {
    const wrapper = mount(base);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('type demo works fine', () => {
    const wrapper = mount(type);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('events demo works fine', () => {
    const wrapper = mount(events);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('maxlength demo works fine', () => {
    const wrapper = mount(maxlength);
    expect(wrapper.element).toMatchSnapshot();
  });
});
