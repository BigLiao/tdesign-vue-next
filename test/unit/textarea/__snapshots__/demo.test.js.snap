// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea base demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
>
  <div
    class="t-textarea"
  >
    <textarea
      class="t-textarea__inner"
      modelvalue=""
      name="description"
      placeholder="请输入描述文案"
    />
    <!---->
    <!---->
  </div>
  <div
    class="t-textarea"
  >
    <textarea
      class="t-textarea__inner"
      modelvalue=""
      name="description"
      placeholder="请输入文案，高度可自适应；autosize=true"
    />
    <!---->
    <!---->
  </div>
  <div
    class="t-textarea"
  >
    <textarea
      class="t-textarea__inner"
      modelvalue=""
      name="description"
      placeholder="请输入文案，高度可自适应，最小3行，最大5行；autosize={minRows: 3, maxRows: 5}"
    />
    <!---->
    <!---->
  </div>
</div>
`;

exports[`Textarea events demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
>
  <div
    class="t-textarea"
  >
    <textarea
      class="t-textarea__inner"
      modelvalue=""
      placeholder="请输入"
    />
    <!---->
    <!---->
  </div>
</div>
`;

exports[`Textarea maxlength demo works fine 1`] = `
<div>
  <div
    class="t-textarea"
  >
    <textarea
      class="t-textarea__inner t-resize-none"
      maxlength="20"
      modelvalue=""
      placeholder="请输入描述文案，文本长度最多20，maxlength=20"
    />
    <!---->
    <span
      class="t-textarea__limit"
    >
      0/20
    </span>
  </div>
  <div
    class="t-textarea"
  >
    <textarea
      class="t-textarea__inner"
      modelvalue=""
      placeholder="请输入描述文案，最多20字符（一个汉字占两个字符长度），maxcharacter=20"
    />
    <span
      class="t-textarea__limit"
    >
      0/20
    </span>
    <!---->
  </div>
</div>
`;

exports[`Textarea type demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
>
  <div
    class="t-textarea"
  >
    <textarea
      class="t-textarea__inner t-is-disabled"
      disabled=""
      placeholder=""
    />
    <!---->
    <!---->
  </div>
  <div
    class="t-textarea"
  >
    <textarea
      class="t-textarea__inner"
      placeholder=""
      readonly=""
    />
    <!---->
    <!---->
  </div>
</div>
`;
